tester/runner_test.go

```diff
@@ -16,6 +16,125 @@ import (
 	"github.com/open-policy-agent/opa/util/test"
 )
 
+func TestRunner_EnableFailureLine(t *testing.T) {
+
+	ctx := context.Background()
+
+	files := map[string]string{
+		"/a_test.rego": `package foo
+			test_a { 
+				true
+				false
+				true
+			}
+			test_b { 
+				false
+				true
+			}`,
+	}
+
+	tests := map[[2]string]struct {
+		wantErr  bool
+		wantFail bool
+		FailRow  int
+	}{
+		{"data.foo", "test_a"}: {false, true, 4},
+		{"data.foo", "test_b"}: {false, true, 8},
+	}
+
+	test.WithTempFS(files, func(d string) {
+		paths := []string{d}
+		modules, store, err := tester.Load(paths, nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+		ch, err := tester.NewRunner().EnableFailureLine(true).SetStore(store).Run(ctx, modules)
+		if err != nil {
+			t.Fatal(err)
+		}
+		var rs []*tester.Result
+		for r := range ch {
+			rs = append(rs, r)
+		}
+		seen := map[[2]string]struct{}{}
+		for i := range rs {
+			k := [2]string{rs[i].Package, rs[i].Name}
+			seen[k] = struct{}{}
+			exp, ok := tests[k]
+			if !ok {
+				t.Errorf("Unexpected result for %v", k)
+			} else if exp.wantErr != (rs[i].Error != nil) || exp.wantFail != rs[i].Fail {
+				t.Errorf("Expected %v for %v but got: %v", exp, k, rs[i])
+			} else if rs[i].FailedAt == nil || rs[i].FailedAt.Location == nil {
+				t.Errorf("Failed line not set")
+			} else if rs[i].FailedAt.Location.Row != exp.FailRow {
+				t.Errorf("Expected Failed Line %v but got: %v", exp.FailRow, rs[i].FailedAt.Location.Row)
+			}
+		}
+		// This makes sure all tests were executed
+		for k := range tests {
+			if _, ok := seen[k]; !ok {
+				t.Errorf("Expected result for %v", k)
+			}
+		}
+	})
+}
+
+func TestRunner_EnableFailureLine_NoRule(t *testing.T) {
```

Comment:
Is there some way this can be folded into the case above? It's a lot of duplication for a single test case.

Reply:
Done